docs/scripting.md

```diff
@@ -260,6 +260,23 @@ module.exports = (robot) ->
     res.send res.random leaveReplies
 ```
 
+## Custom Listeners
+
+While the above helpers cover most of the functionality the average user needs (hear, respond, enter, leave, topic), sometimes you would like to have very specialized matching logic for listeners. If so, you can use `listen` to specify a custom match function instead of a regular expression.
+
+```coffeescript
+module.exports = (robot) ->
+  robot.listen(
+    (message) ->
+      # This function must return a truthy value if the listener callback should be executed.
```

Comment:
If you look at the other documentation in this file, it typically is a 'real' example, as in it's something you can copy and paste to run and it'd do something. It'd be good to have that here too, in addition to the comments.